pycbc/workflow/injection.py

```diff
@@ -50,19 +51,67 @@ def veto_injections(workflow, inj_file, veto_file, veto_name, out_dir, tags=None
+    """Compute optimal SNR for injections"""
+    current_retention_level = Executable.ALL_TRIGGERS
+
+    def create_node(self, workflow, inj_file, precalc_psd_files, group_str):
+        node = Node(self)
+        node.add_input_opt('--input-file', inj_file)
+        node.add_opt('--injection-fraction-range', group_str)
+        node.add_input_list_opt('--time-varying-psds', precalc_psd_files)
+        node.new_output_file_opt(workflow.analysis_time, '.xml',
+                                 '--output-file')
+        return node
+
+
 def compute_inj_optimal_snr(workflow, inj_file, precalc_psd_files, out_dir,
                             tags=None):
     "Set up a job for computing optimal SNRs of a sim_inspiral file."
     if tags is None:
         tags = []
 
-    node = Executable(workflow.cp, 'optimal_snr', ifos=workflow.ifos,
-                      out_dir=out_dir, tags=tags).create_node()
-    node.add_input_opt('--input-file', inj_file)
-    node.add_input_list_opt('--time-varying-psds', precalc_psd_files)
-    node.new_output_file_opt(workflow.analysis_time, '.xml', '--output-file')
-    workflow += node
-    return node.output_files[0]
+    try:
+        factor = int(workflow.cp.get_opt_tags('workflow-optimal-snr',
+                                              'parallelization-factor',
+                                              tags))
+    except ConfigParser.Error:
+        factor = 1
+
+    if factor == 1:
+        # parallelization factor not given - default to single optimal snr job
+        opt_snr_exe = PyCBCOptimalSNRExecutable(workflow.cp, 'optimal_snr',
+                                                ifos=workflow.ifos,
+                                                out_dir=out_dir, tags=tags)
+        node = opt_snr_exe.create_node(workflow, inj_file,
+                                       precalc_psd_files, '0/1')
+        workflow += node
+
+        return node.output_files[0]
+
+    opt_snr_split_files = []
+    for i in range(factor):
+        group_str = '%s/%s' % (i, factor)
+        opt_snr_exe = PyCBCOptimalSNRExecutable(workflow.cp, 'optimal_snr',
+                                                ifos=workflow.ifos,
+                                                out_dir=out_dir,
+                                                tags=tags + [str(i)])
+        node = opt_snr_exe.create_node(workflow, inj_file, precalc_psd_files,
+                                       group_str)
+        opt_snr_split_files += [node.output_files[0]]
+        workflow += node
+
+    llwadd_exe = LigolwAddExecutable(workflow.cp, 'optimal_snr_merge',
+                                     ifos=workflow.ifos, out_dir=out_dir,
+                                     tags=tags)
+    merge_node = llwadd_exe.create_node(workflow.analysis_time,
+                                        opt_snr_split_files,
+                                        use_tmp_subdirs=False)
+    merge_node.add_opt('--ilwdchar-compat')
```

Comment:
Would it be better to keep this option in the config file? We may run the workflow under an environment with an older version of `ligolw_add` that does not yet have the option.

Reply:
that sounds fair

Reply:
In fact it seems even fairly old CVMFS PyCBC envs do have the option, so this actually seems safer than I thought. Still, presumably at some point the option will be removed from `ligolw_add`...